volume/fs.go

```diff
@@ -0,0 +1,90 @@
+package volume
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/puppetlabs/wash/activity"
+	"github.com/puppetlabs/wash/plugin"
+)
+
+// FS presents a view of the filesystem of an Execable resource.
+type FS struct {
+	plugin.EntryBase
+	executor plugin.Execable
+}
+
+// NewFS creates a new FS entry with the given name, using the supplied executor to satisfy volume
+// operations.
+func NewFS(name string, executor plugin.Execable) *FS {
+	fs := &FS{EntryBase: plugin.NewEntry(name), executor: executor}
+	// Caching handled in List.
+	fs.DisableCachingFor(plugin.ListOp)
+	return fs
+}
+
+// List will attempt to list the filesystem of an Execable resource (the executor). It will list
+// a directory tree based on supplied configuration (defaulting to `/var/log` if not specified).
+func (d *FS) List(ctx context.Context) ([]plugin.Entry, error) {
+	return List(ctx, d, "")
+}
+
+func consumeExec(ctx context.Context, result plugin.ExecResult) (*bytes.Buffer, error) {
+	var buf bytes.Buffer
+	for chunk := range result.OutputCh {
+		if chunk.Err != nil {
+			activity.Record(ctx, "Error on exec: %v", chunk.Err)
+			return nil, chunk.Err
+		} else if chunk.StreamID == 0 {
+			activity.Record(ctx, "Stdout: %v", chunk.Data)
+			fmt.Fprint(&buf, chunk.Data)
+		} else {
+			activity.Record(ctx, "Stderr: %v", chunk.Data)
+		}
+	}
+
+	exitcode, err := result.ExitCodeCB()
+	if err != nil {
+		activity.Record(ctx, "Error exiting exec: %v", err)
```

Comment:
I think these errors should include the failed command. Otherwise, it's not obvious what failed to exec without looking through the code.

Reply:
I guess the activity records that info, so maybe it's not that important.

Reply:
Clarified it a bit anyway.